sailthru/sailthru_content/services/catalog_api_service.py

```diff
@@ -63,25 +63,18 @@ def get_courses(self):
         logger.debug('Get Courses called')
         return self._get_resource_from_api(self.api_client.courses(), COURSES_PAGE_SIZE, marketable=1)
 
-    def get_searchable_course_keys(self):
+    def get_searchable_course_run_keys(self):
         logger.debug('Get Searchable Courses called')
         searchable_course_keys = []
-        searchable_courses = self._get_resource_from_search_api(self.api_client.search.all.facets())
-        for course in searchable_courses:
-            if course.get('key') and course.get('content_type') != 'program':
-                searchable_course_keys.append(course['key'])
+        searchable_course_runs = self._get_resource_from_search_api(
+            self.api_client.search.all.facets(), content_type='courserun'
```

Comment:
Got it.  After leaving you the comment, I did some digging.  Take a look at the course_run endpoint:
https://prod-edx-discovery.edx.org/api-docs/#!/api/Course_Run_list

You'll see in the swagger documentation that it provides you the 'marketable' flag.  I think you should scrap the dictionary/key approach and just use this endpoint with the marketable flag, is there a reason not to do so?